batchglm/train/numpy/glm_nb/modelContainer.py

```diff
@@ -1,47 +1,19 @@
-import logging
-from typing import Union
+from typing import Callable, Union
 
-import dask.array
+import dask
 import numpy as np
-import scipy.sparse
-import scipy.special
-import sparse
+import scipy
 
-from .external import InputDataGLM, Model, ModelIwls
-from .processModel import ProcessModel
+from .external import BaseModelContainer
 
-logger = logging.getLogger(__name__)
 
-
-class ModelIwlsNb(ModelIwls, Model, ProcessModel):
-
-    """
-    Class for maintaining state of Negative Binomial IWLS updates.
-
-    Attributes
-    ----------
-    compute_mu : bool
-    compute_r : bool
-    """
-
-    compute_mu: bool
-    compute_r: bool
-
-    def __init__(
-        self,
-        input_data: InputDataGLM,
-        model_vars,
-        compute_mu,
-        compute_r,
-    ):
-        self.compute_mu = compute_mu
-        self.compute_r = compute_r
-
-        super(Model, self).__init__(input_data=input_data)
-        ModelIwls.__init__(self=self, model_vars=model_vars)
+class ModelContainer(BaseModelContainer):
+    @property
+    def fim_weight(self):
+        raise NotImplementedError("This method is currently unimplemented as it isn't used by any built-in procedures.")
```

Comment:
So this message is should be about NB specifically right, like "any built-in procedures of IWLS for NB"?

Reply:
Not sure what you mean. I just added this method because mypy would complain it isn't implemented but it is abstract in the parent class. Maybe we can get rid of it entirely?

Reply:
Not sure.  In theory it will be needed, so I don't think removing it completely makes sense.  I was just commenting on the question of why we don't need this property for negative binomial.